beacon-chain/core/blocks/block_operations_test.go

```diff
@@ -0,0 +1,146 @@
+	registry := []*pb.ValidatorRecord{}
+	currentSlot := uint64(0)
+
+	if _, err := ProcessProposerSlashings(
+		registry,
+		slashings,
+		currentSlot,
+	); err == nil {
+		want := "number of proposer slashings exceeds threshold"
+		t.Errorf("Expected %v, received nil", want)
+	}
+	currentSlot++
+
+	// We now test the case with unmatched slot numbers.
+	slashings = []*pb.ProposerSlashing{
+		{
+			ProposerIndex: 0,
+			ProposalData_1: &pb.ProposalSignedData{
+				Slot: 1,
+			},
+			ProposalData_2: &pb.ProposalSignedData{
+				Slot: 0,
+			},
+		},
+	}
+
+	if _, err := ProcessProposerSlashings(
+		registry,
+		slashings,
+		currentSlot,
+	); err == nil {
+		want := "slashing proposal data slots do not match: 1, 0"
+		t.Errorf("Expected %v, received nil", want)
+	}
+	currentSlot++
+
+	// We now test the case with unmatched shard IDs.
+	slashings = []*pb.ProposerSlashing{
+		{
+			ProposerIndex: 0,
+			ProposalData_1: &pb.ProposalSignedData{
+				Slot:  1,
+				Shard: 0,
+			},
+			ProposalData_2: &pb.ProposalSignedData{
+				Slot:  1,
+				Shard: 1,
+			},
+		},
+	}
+
+	if _, err := ProcessProposerSlashings(
+		registry,
+		slashings,
+		currentSlot,
+	); err == nil {
+		want := "slashing proposal data shards do not match: 0, 1"
+		t.Errorf("Expected %v, received nil", want)
+	}
+	currentSlot++
+
+	// We now test the case with unmatched block hashes.
+	slashings = []*pb.ProposerSlashing{
+		{
+			ProposerIndex: 0,
+			ProposalData_1: &pb.ProposalSignedData{
+				Slot:        1,
+				Shard:       0,
+				BlockHash32: []byte{0, 1, 0},
+			},
+			ProposalData_2: &pb.ProposalSignedData{
+				Slot:        1,
+				Shard:       0,
+				BlockHash32: []byte{1, 1, 0},
+			},
+		},
+	}
+
+	if _, err := ProcessProposerSlashings(
+		registry,
+		slashings,
+		currentSlot,
+	); err == nil {
+		want := fmt.Sprintf(
+			"slashing proposal data block hashes do not match: %x, %x",
+			[]byte{0, 1, 0}, []byte{1, 1, 0},
+		)
+		t.Errorf("Expected %v, received nil", want)
+	}
+}
+
+func TestCorrectlyProcessProposerSlashings(t *testing.T) {
+	// We test the case when data is correct and verify the validator
+	// registry has been updated.
+	registry := []*pb.ValidatorRecord{
+		{
+			Status:                 uint64(params.ExitedWithPenalty),
+			LatestStatusChangeSlot: 0,
+		},
+		{
+			Status:                 uint64(params.ExitedWithPenalty),
+			LatestStatusChangeSlot: 0,
+		},
+	}
+	slashings := []*pb.ProposerSlashing{
+		{
+			ProposerIndex: 1,
+			ProposalData_1: &pb.ProposalSignedData{
+				Slot:        1,
+				Shard:       1,
+				BlockHash32: []byte{0, 1, 0},
+			},
+			ProposalData_2: &pb.ProposalSignedData{
+				Slot:        1,
+				Shard:       1,
+				BlockHash32: []byte{0, 1, 0},
+			},
+		},
+	}
+	currentSlot := uint64(1)
+
+	registry, err := ProcessProposerSlashings(
+		registry,
+		slashings,
+		currentSlot,
+	)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	if registry[1].Status != uint64(params.ExitedWithPenalty) {
```

Comment:
This should actually use StatusCode: https://github.com/prysmaticlabs/prysm/blob/b4f8c0afaf4c3d0de48f22c47cb30d4b80e44735/proto/beacon/p2p/v1/types.proto#L105

Reply:
Ignore this, `status_code` is now deprecated and `status` is correct. #1082